docs/control-structures.rst

```diff
@@ -20,6 +20,45 @@ there is in C and JavaScript, so ``if (1) { ... }`` is *not* valid Solidity.
 
 .. index:: ! function;call, function;internal, function;external
 
+Input Parameters and Return Parameters
+======================================
+
+Like in Javascript and in C, functions may take parameters as input;
+unlike in Javascript and C, they may also return arbitrary number of
+parameters as output. The input parameters are declared type followed
+by name (though unused input parameters do not need names).
+For example, suppose we want our contract to
+accept one kind of external calls with two integers, we would write
+something like::
+
+    contract Simple {
+        function taker(uint _a, uint _b) {
+            // do something with _a and _b.
+        }
+    }
+
+Returning information is equally simple using the same syntax with the
+``returns`` keyword. For example, suppose we wished to return two results:
+the sum and the product of the two given integers, then we would
+write::
+
+    contract Simple {
+        function arithmetics(uint _a, uint _b) returns (uint o_sum, uint o_product) {
+            o_sum = _a + _b;
+            o_product = _a * _b;
```

Comment:
Now I believe this is an abbreviation, but not returning a tuple, because the following does not work:

```
contract C {

function f () returns (uint a, uint b) {
    var pair = (3, 4);
    return pair;
}

}
```